### ISSUE TITLE
[new design] missing dropdown-styles for headerbar

### ISSUE THREAD

**Issue by svom** [joomla/40-backend-template#386]
Stylings for the Headerbar dropdowns are missing. See screenshots of proposal attached.

![bildschirmfoto 2018-04-16 um 16 08 45](https://user-images.githubusercontent.com/14434412/38814041-cf01f79e-4190-11e8-953e-9203e7cad230.png)
![bildschirmfoto 2018-04-16 um 16 08 51](https://user-images.githubusercontent.com/14434412/38814042-cf1cc40c-4190-11e8-8b4e-75240b52af28.png)
![bildschirmfoto 2018-04-16 um 16 08 34](https://user-images.githubusercontent.com/14434412/38814043-cf363202-4190-11e8-9953-a99f0174a3cd.png)
[20180302-172412.pdf](https://github.com/joomla/40-backend-template/files/1915602/20180302-172412.pdf)

**Comment by lakhanmandloi** [comment]
@svom this looks nice to me. :+1: 

One more thing i would like to add in this - 

In mobile devices / smaller screens - the headerbar covers a lot of area, i think we should reduce it by hiding less important text. I mean we should reduce 2 rows to single row. This headerbar is a sticky item and would be bad if it covers a lot of area. 

**Current**
![current](https://user-images.githubusercontent.com/30073416/40483897-5617f04a-5f77-11e8-8557-843489aaf1f3.png)

**Proposal**
![proposal](https://user-images.githubusercontent.com/30073416/40483914-67869782-5f77-11e8-9a6b-7bd9e9fb5e04.png)

Note - pls ignore the icons in this proposal image, this image represents more about reducing the headerbar size in smaller devices. The icons will come from the @svom proposed designs. 

Pls let me know your opinion on same - @svom @ciar4n @dgrammatiko 
Once this is finalised i will start contributing on this issue.  

Thanks!

**Comment by ciar4n** [comment]
+1 from me!

I would add we could also echo the title in the main container and have it only displayed on small screen devices while hiding the title in the header..

![image](https://user-images.githubusercontent.com/2803503/40484846-cc719ab6-5f54-11e8-9b75-e25c5522f9da.png)

**Comment by ciar4n** [comment]
@svom A bit more to consider with this but should be possible.

@lakhanmandloi If you could work on the dropdown styles for the header actions that would be great! We can consider smaller screens separately at a later date. Thank you!

**Comment by dgrammatiko** [comment]
@lakhanmandloi please wait for me to provide the custom element for the dropdown. We're **not** gonna use the bootstrap dropdown anymore 😉

**Comment by lakhanmandloi** [comment]
> I would add we could also echo the title in the main container and have it only displayed on small screen devices while hiding the title in the header..
Agree. that's nice approach.:+1:  and it seems @svom has already covered this in proposal.

> But in general we had the idea (and showed the idea) to place the navigation as a sticky toolbar at the bottom of the device on smaller devices (mostly smartphones).
> In  this case it's easier for the user to reach the navigation (due to the anatomy of our hand).

Agree, bringing the main navigation to the bottom in smaller devices is a very nice idea from the ux perspective, and will give main component area content more space to expand. :+1: 

but i have few questions:
-  what if someone adds more menu-items on level 1 of main navigation. will there be horizontal scrollbar for same?
- how will level 2 navigation be displayed? can you please propose a expanded menu design for same?
- from this proposal it seems messages and notifications will go inside the drop-down of account (in smaller devices). But i think we should keep them outside and make them clearly visible. I mean these 2 items are important and should not be hidden somewhere. 

> please wait for me to provide the custom element for the dropdown. We're not gonna use the bootstrap dropdown anymore . 
Okay i will wait. :) do let me know once ready. 

Do let me know if some other issues which are very much clear to do and needs contribution. I happy in contributing to the Joomla Community. :)

Thank you again. :)

**Comment by svom** [comment]
> but i have few questions:
> 
> what if someone adds more menu-items on level 1 of main navigation. will there be horizontal scrollbar for same?

Could be one opportunity, another opportunity which comes to my mind is to give control about which navitems should be displayed on mobile devices. Actually I guess that the needs of the people are different regarding mobile and desktop navigation, because even in 2018 or 19 most of the people will administrate the page on desktop-devices. Therefore it would be great to have a customizable navbar - so people can refer to their needs. Nevertheless a horizontal scrollbar (actually you don't even need a scrollbar, but only a hint to show that it's possible to drag the navigation) would be a working solution.

> how will level 2 navigation be displayed? can you please propose a expanded menu design for same?
I will.

> from this proposal it seems messages and notifications will go inside the drop-down of account (in smaller devices). But i think we should keep them outside and make them clearly visible. I mean these 2 items are important and should not be hidden somewhere.

We can use a small visual indicator to show that there are notifcations and stuff (like facebook messenger and co are doing).

**Comment by lakhanmandloi** [comment]
> Could be one opportunity, another opportunity which comes to my mind is to give control about which navitems should be displayed on mobile devices. Actually I guess that the needs of the people are different regarding mobile and desktop navigation, because even in 2018 or 19 most of the people will administrate the page on desktop-devices. Therefore it would be great to have a customizable navbar - so people can refer to their needs. Nevertheless a horizontal scrollbar (actually you don't even need a scrollbar, but only a hint to show that it's possible to drag the navigation) would be a working solution.

I absolutely agree with your points. From "customizable navbar" do you mean drag and drop reshuffling ? If yes, +100  :+1: . That will be great ux improvement. 
In general, what if we apply Drag and drop reshuffling concept at some other places too. Right now the reshuffling of items is limited to items list only, like in modules, menus, articles list etc.
What if administrator users are allowed to - 
- reshuffle modules within the position placeholder. Eg - Admin dashboard , administrator are allowed to reshuffle the modules by simply drag and drop and not limited to module management.
- Main navigation - adminstrator are allowed to reshuffle admin menu sequence on any page rather than going to Menu Management. And the same sequence applies to mobile devices.

And i am sure, by this addition 
- we will be able to add a unique UX improvement which hardly any other CMS might be having. 
- this will bring more of a native app feeling so more mobile friendly CMS.
- reduces efforts for small config changes. 

Some inspirations - 
- http://demo.thedevelovers.com/dashboard/klorofilpro-v1.6/html/ui-dragdrop-panel.html
- http://gridstackjs.com/ ( Probably we can mould this in BS4)

What do you think ?

> We can use a small visual indicator to show that there are notifcations and stuff (like facebook messenger and co are doing).

Agree. Can you pls include this in design so that it is more clear.?
